app/models/mappers/check-history.js

```diff
@@ -191,13 +191,19 @@ const getActivityLabelFromEvent = (event) => {
     return 'NOT YET DEFINED'
   }
 
-  if (event.activity?.activityType === 'received' && event.activity?.activityLabel.toLowerCase().includes('received')) {
-    return `${event.activity?.activityLabel}`
-  }
-
   if (event.activity?.activityType) {
-    return `${event.activity?.activityLabel} ${event.activity?.activityType}`
+    // Special case - for Form 2 sent
+    if (event.activity.activityType === 'sent' && event.activity?.activityLabel.startsWith('Form 2 from ')) {
+      return `${event.activity.activityLabel} requested`
+    }
+
+    if (event.activity.activityType === 'received' && event.activity?.activityLabel.toLowerCase().includes('received')) {
+      return `${event.activity.activityLabel}`
+    }
+
+    return `${event.activity?.activityLabel} ${event.activity.activityType}`
   }
+
```

Comment:
not going to let this hang things up, and have done in some occassions, but try to get away from nested ifs if at all possible.